src/main/java/com/easypost/service/ShipmentService.java

```diff
@@ -461,4 +461,24 @@ public List<EstimatedDeliveryDate> retrieveEstimatedDeliveryDate(final String id
                 EstimatedDeliveryDateResponse.class, client);
         return response.getRates();
     }
+
+    /**
+     * Retrieve a recommended ship date for an existing Shipment via the Precision Shipping API,
+     * based on a specific desired delivery date.
+     *
+     * @param id              The id of the shipment.
+     * @param desiredDeliveryDate The desired delivery date.
+     * @return EstimatedDeliveryDate object.
+     * @throws EasyPostException When the request fails.
+     */
+    public List<EstimatedDeliveryDate> recommendShipDate(final String id, final String desiredDeliveryDate)
+            throws EasyPostException {
+        HashMap<String, Object> params = new HashMap<>();
+        params.put("desired_delivery_date", desiredDeliveryDate);
+        String endpoint = "shipments/" + id + "/smartrate/precision_shipping";
+
+        EstimatedDeliveryDateResponse response = Requestor.request(RequestMethod.GET, endpoint, params,
```

Comment:
How does this deserialize into the new models? This would make a `EstimatedDeliveryDateResponse`, which has a list of `EstimateDeliveryDate` object, which each have `Rate` and `TimeInTransit` properties. Where do the new models come into play in the serialization process?

Reply:
Good catch. I have created the new models for this response